lib/spack/spack/util/compression.py

```diff
@@ -602,13 +691,16 @@ def extension_from_path(path):
     return None
 
 
-def strip_last_extension(path):
-    """Strips last supported archive extension from path"""
-    if path:
-        for ext in ALLOWED_SINGLE_EXT_ARCHIVE_TYPES:
-            mod_path = check_and_remove_ext(path, ext)
-            if mod_path != path:
-                return mod_path
+def strip_last_extension(path, ext=None):
```

Comment:
I think `strip_compression_extension` would be a clearer name (IMO a function called `strip_last_extension` would transform `foo.tgz -> foo`).

Reply:
Good point! Will amend.